aptos-move/aptos-release-builder/src/components/framework.rs

```diff
@@ -26,9 +31,21 @@ pub fn generate_upgrade_proposals(
 
     let mut result: Vec<(String, String)> = vec![];
 
-    let mut root_path = std::path::Path::new(env!("CARGO_MANIFEST_DIR")).to_path_buf();
-    root_path.pop();
-    root_path.pop();
+    let temp_root_path = TempPath::new();
+    temp_root_path.create_as_dir()?;
+
+    let commit_info = if let Some(revision) = &config.git_hash {
```

Comment:
Can you add some comments to explain what this is trying to do?

Reply:
Yes. The goal here is we want to generate the aptos framework located at a given commit hash. This would allow us to build the release on top of main branch if needed